core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -845,6 +852,13 @@ private[spark] object Utils extends Logging {
     customHostname.getOrElse(localIpAddressHostname)
   }
 
+  /**
+   * Get the local machine's URI.
+   */
+  def localHostURI(): String = {
```

Comment:
It is named ```localHostURI``` in compliance with ```localHostName``` function name.
The main idea is to use ip if and only if we don't have ```customHostname``` ( like in ```localHostName``` ).